integration-libs/cdc/root/service/cdc-js.service.ts

```diff
@@ -610,6 +614,48 @@ export class CdcJsService implements OnDestroy {
     });
   }
 
+  getSiteConsentDetails(): Observable<CdcSiteConsentTemplate> {
+    let baseSite: string = this.getCurrentBaseSite();
+    let javascriptURL: string = this.getJavascriptUrlForCurrentSite(baseSite);
+    let queryParams = new URLSearchParams(
+      javascriptURL.substring(javascriptURL.indexOf('?'))
+    );
+    let siteApiKey: string | null = queryParams.get('apikey');
+
+    return this.invokeAPI('accounts.getSiteConsentDetails', {
+      apiKey: siteApiKey,
+    });
+  }
+
+  protected currentUID: string | undefined;
+
+  getUserConsentPreferences(
```

Comment:
This `getUserConsentPreferences` can be avoided. In https://github.com/SAP/spartacus/blob/3ff03d7d421fe8756bd8d82e4d8d85037a07d96e/integration-libs/cdc/root/service/cdc-js.service.ts#L124 `preferences` can be included as well.

So the preferences can be obtained on login itself.